### PR TITLE
API Docs Fixes

### PR DESCRIPTION
**Pull request by aliabd**
A bunch of fixes to the API docs: 
* Fixes the docs for audio. related discussion: https://github.com/gradio-app/gradio/discussions/2739
* Fixes the type/description for all components that have different input/output types 
* Fixes example issues for all components 
* Small fixes to some errors 
* Fixes indentation 
* To simplify things only added slight syntax highlighting to important parts of code snippets. We can use this until we have a code component. 
* Added copy buttons to code snippets

closes: #2733 
closes: #1581

### CODE REVIEW

File: ui/packages/app/src/api_docs/ResponseObject.svelte [hunk 1]
```diff
@@ -26,7 +26,7 @@
 		&nbsp;&nbsp;"data": [
 		<br />
 		{#each dependency.outputs as component_id, component_index}
-			&nbsp;&nbsp;&nbsp;{#if dependency_outputs[dependency_index][component_index] !== undefined}
+			&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;{#if dependency_outputs[dependency_index][component_index] !== undefined}
```
Comment (abidlabs) [inline ui/packages/app/src/api_docs/ResponseObject.svelte:29]: Probably a better way to do this? Idk lol

File: ui/packages/app/src/components/Audio/index.ts [hunk 2]
```diff
@@ -2,8 +2,15 @@ export { default as Component } from "./Audio.svelte";
...
-	type: "{ name: string; data: string }",
-	description: "audio data as base64 string",
+	type: {
+		input_payload: "{ name: string; data: string }",
+		response_object: "{ name: string; data: string, is_file: boolean }"
+	},
+	description: {
+		input_payload: "audio data as object with filename and base64 string",
+		response_object:
+			"audio data as object with filename. `{DEMO URL}/file={name}` hosts the audio"
```
Comment (abidlabs) [inline ui/packages/app/src/components/Audio/index.ts:12]: I think we can reword this to make it a little clearer:
```suggestion
			"object that includes path to audio file. Can be accessed at the URL: `{DEMO URL}/file={name}`"
```
Would it be possible by any chance to replace {DEMO URL} with the actual URL of the demo? That would be pretty cool

File: ui/packages/app/src/components/File/index.ts [hunk 3]
```diff
@@ -2,8 +2,16 @@ export { default as Component } from "./File.svelte";
...
-	description: "file name and base64 data as an object",
+	type: {
+		input_payload: "{ name: string; data: string }",
+		response_object:
+			"{ orig_name: string; name: string, size: number, data: string, is_file: boolean}"
+	},
+	description: {
+		input_payload: "object with file name and base64 data",
+		response_object:
+			"file data as object with filename. `{DEMO URL}/file={name}` hosts the data"
```
Comment (abidlabs) [inline ui/packages/app/src/components/File/index.ts:13]: Same comment as above about `Audio` filepaths. I think we can make this a little clearer.

**Comment by abidlabs** [comment]
Really really cool @aliabd! Left a few nits above. Would be good to have someone else review all this nice svelte

My only other suggestion is do you think we can include copy-paste buttons for the code snippet? That would be nice and we can close this issue too: https://github.com/gradio-app/gradio/issues/1581 (also fine to leave for a future PR)

Otherwise LGTM!

File: ui/packages/app/src/api_docs/ResponseObject.svelte (hunk 1 above)
Comment (pngwn) [reply ui/packages/app/src/api_docs/ResponseObject.svelte:29]: Can't we just apply `whitespace: pre` or something?

File: ui/packages/app/src/components/Audio/index.ts (hunk 2 above)
Comment (pngwn) [reply ui/packages/app/src/components/Audio/index.ts]: This is knocking around somewhere. Maybe `base_api`, which we pass to the API, would do the trick.

File: ui/packages/app/src/api_docs/ResponseObject.svelte (hunk 1 above)
Comment (aliabd) [reply ui/packages/app/src/api_docs/ResponseObject.svelte]: I did this with margins 

Screen Shot 2023-02-22 at 6 53 10 PM

File: ui/packages/app/src/components/Audio/index.ts (hunk 2 above)
Comment (aliabd) [reply ui/packages/app/src/components/Audio/index.ts]: Reworded it and replaced with actual URL 


https://user-images.githubusercontent.com/9021060/220811737-af7ccf7c-97b8-48ec-b5e5-28a0cf506877.mov

File: ui/packages/app/src/components/File/index.ts (hunk 3 above)
Comment (aliabd) [reply ui/packages/app/src/components/File/index.ts]: Yes made all changes to audio to file and video too

**Comment by aliabd** [comment]
Thanks @abidlabs! Yes @pngwn please give me a brutal review as I feel like some of my code is hacky 🙏 

I added copy buttons to the code snippets: 


https://user-images.githubusercontent.com/9021060/220812260-6a9661d5-ab77-4612-97a4-017457b08d99.mov

**Comment by abidlabs** [comment]
Will leave the brutal feedback to @pngwn because fixes really good to me! This will be very handy for the gradio client as I think we'll pull these descriptions to provide users instructions on how to query a speciic API

**Review by abidlabs — APPROVED** [review]

**Review by pngwn — APPROVED** [review]
I think this looks good now. We can always tweak the formatting approach in a follow-up, i'll do a little testing and see if i can come up with something nicer but i think this is good to merge now.
